### PR TITLE
ci: Deploy iOS-Swift to Testflight

### PR DESCRIPTION
**Pull request by philipphofmann**
## :scroll: Description

Every commit on master triggers a deploy of the iOS-Swift Sample app to Testflight. See it in action here: https://github.com/getsentry/sentry-cocoa/actions?query=workflow%3A%22Upload+to+Testflight%22

## :bulb: Motivation and Context

We want to be able to test a real-world scenario use of our SDK on a regular basis.

## :green_heart: How did you test it?
Downloading the build via Testflight.

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I updated the CHANGELOG if needed
- [x] I updated the docs if needed
- [x] Review from the native team if needed
- [x] No breaking changes

### CODE REVIEW

File: .github/workflows/testflight.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      # https://github.com/actions/virtual-environments/issues/1820#issuecomment-719549887
+      
+      # Sometimes the keychain already exists
+      - name: Delete local keychain
+        continue-on-error: true
+        run: security delete-keychain $FASTLANE_KEYCHAIN_NAME
+          
+      - name: Create local keychain
+        run: |
+          security create-keychain -p ${{ secrets.FASTLANE_KEYCHAIN_PASSWORD }} $FASTLANE_KEYCHAIN_NAME
```
Comment (bruno-garcia) [inline .github/workflows/testflight.yml:33]: Are these passwords only for the specific test app? What can one do with these secrets?

**Review by bruno-garcia — CHANGES_REQUESTED** [review]

File: .github/workflows/testflight.yml (hunk 1 above)
Comment (philipphofmann) [reply .github/workflows/testflight.yml:33]: Maybe we should rather discuss in private what you can do with these secrets.

Comment (bruno-garcia) [reply .github/workflows/testflight.yml:33]: Maybe lets have a _secret_ meeting 🤫

File: .github/workflows/testflight.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
+name: Upload to Testflight
+on:
+  push:
+    branches:
+      - master
+      - ci/** # Todo remove before merging
```
Comment (marandaneto) [inline .github/workflows/testflight.yml:6]: H: remove before merging :D

File: .github/workflows/testflight.yml (hunk 1 above)
Comment (marandaneto) [reply .github/workflows/testflight.yml:33]: let's do it over signal :D

**Review by marandaneto — APPROVED** [review]
LGTM
we should definitely talk about the `secrets` before going ahead though.

**Comment by philipphofmann** [comment]
Upload fails because of privacy statement not accepted. According to [this issue](https://github.com/fastlane/fastlane/issues/12577), all users in TestFlight need to accept the privacy statement. I'm currently pinging everyone and will try again soon.

**Pull request merged**